wfe/wfe.go

```diff
@@ -329,6 +331,54 @@ func (wfe *WebFrontEndImpl) handleKey(
 	}
 }
 
+func (wfe *WebFrontEndImpl) handleCertStatusBySerial(
+	ctx context.Context,
+	response http.ResponseWriter,
+	request *http.Request) {
+
+	serialStr := strings.TrimPrefix(request.URL.Path, certStatusBySerial)
+	serial := big.NewInt(0)
+	if _, ok := serial.SetString(serialStr, 16); !ok {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	var status string
+	var cert *core.Certificate
+	var rcert *core.RevokedCertificate
+	if cert = wfe.db.GetCertificateBySerial(serial); cert != nil {
+		status = "Valid"
+	}
+	if rcert = wfe.db.GetRevokedCertificateBySerial(serial); rcert != nil {
```

Comment:
How do you feel about:

1. Flipping the order of these checks so that `wfe.db.GetRevokedCertificateBySerial` is checked first
2. Making the other clause an `else if`.

My reasoning for the first change is that I suspect more often Pebble will be asked about cert's that are expected to be revoked and so we can save some of the linear search time by doing that first. There's also likely to be less certificates to scan linearly in the revoked slice than the overall issued slice.

Reply:
Also, usually the set of revoked certs is much smaller than the set of non-revoked certs, so this makes even more sense.